aspnetcore/fundamentals/startup/includes/startup56.md

```diff
@@ -0,0 +1,188 @@
+
+`IStartupFilter` implements <xref:Microsoft.AspNetCore.Hosting.StartupBase.Configure%2A>, which receives and returns an `Action<IApplicationBuilder>`. An <xref:Microsoft.AspNetCore.Builder.IApplicationBuilder> defines a class to configure an app's request pipeline. For more information, see [Create a middleware pipeline with IApplicationBuilder](xref:fundamentals/middleware/index#create-a-middleware-pipeline-with-iapplicationbuilder).
+
+Each `IStartupFilter` can add one or more middlewares in the request pipeline. The filters are invoked in the order they were added to the service container. Filters may add middleware before or after passing control to the next filter, thus they append to the beginning or end of the app pipeline.
+
+The following example demonstrates how to register a middleware with `IStartupFilter`. The `RequestSetOptionsMiddleware` middleware sets an options value from a query string parameter:
+
+[!code-csharp[](~/fundamentals/startup/3.0_samples/StartupFilterSample/RequestSetOptionsMiddleware.cs?name=snippet1)]
+
+The `RequestSetOptionsMiddleware` is configured in the `RequestSetOptionsStartupFilter` class:
+
+[!code-csharp[](~/fundamentals/startup/3.0_samples/StartupFilterSample/RequestSetOptionsStartupFilter.cs?name=snippet1&highlight=7)]
+
+The `IStartupFilter` is registered in the service container in <xref:Microsoft.AspNetCore.Hosting.StartupBase.ConfigureServices*>.
+
+[!code-csharp[](~/fundamentals/startup/3.0_samples/StartupFilterSample/Program.cs?name=snippet&highlight=19-20)]
+
+When a query string parameter for `option` is provided, the middleware processes the value assignment before the ASP.NET Core middleware renders the response.
+
+Middleware execution order is set by the order of `IStartupFilter` registrations:
+
+* Multiple `IStartupFilter` implementations may interact with the same objects. If ordering is important, order their `IStartupFilter` service registrations to match the order that their middlewares should run.
+* Libraries may add middleware with one or more `IStartupFilter` implementations that run before or after other app middleware registered with `IStartupFilter`. To invoke an `IStartupFilter` middleware before a middleware added by a library's `IStartupFilter`:
+
+  * Position the service registration before the library is added to the service container.
+  * To invoke afterward, position the service registration after the library is added.
+
+## Add configuration at startup from an external assembly
+
+An <xref:Microsoft.AspNetCore.Hosting.IHostingStartup> implementation allows adding enhancements to an app at startup from an external assembly outside of the app's `Startup` class. For more information, see <xref:fundamentals/configuration/platform-specific-configuration>. `IHostingStartup` works with <xref:Microsoft.AspNetCore.Builder.WebApplicationBuilder>.
```

Comment:
It doesn't make sense to mention `WebApplicationBuilder` in the .NET 5 docs. I also don't see why the .NET 6 and .NET 7 startup docs should be any different from each other.